drivers/hmis/app/models/hmis/file.rb

```diff
@@ -35,11 +35,30 @@ def data_source
   scope :nonconfidential, -> { where(confidential: [false, nil]) }
 
   scope :viewable_by, ->(user) do
-    view_scope = where(client_id: Hmis::Hud::Client.with_access(user, :can_view_any_nonconfidential_client_files, :can_view_any_confidential_client_files))
-    # view_scope = view_scope.nonconfidential unless user.can_view_any_confidential_client_files?
-    edit_scope = user.can_manage_own_client_files? ? where(user_id: user.id) : none
-
-    view_scope.or(edit_scope)
+    # NOTE: it's okay that confidential files are included in this scope even if the user
+    # doesn't have permission to read the file. Users can see the existence of confidential
+    # files but they can't read them. Reference:
+    # https://www.pivotaltracker.com/n/projects/2591838/stories/185293913
+    client_scope = Hmis::Hud::Client
+      .viewable_by(user)
+      .with_access(user, :can_view_any_nonconfidential_client_files, :can_view_any_confidential_client_files)
+    enrollment_scope = Hmis::Hud::Enrollment
+      .viewable_by(user)
+      .with_access(user, :can_view_any_nonconfidential_client_files, :can_view_any_confidential_client_files)
+
+    case_statement = Arel::Nodes::Case.new
+      .when(arel_table[:enrollment_id].not_eq(nil))
+      .then(arel_table[:enrollment_id].in(enrollment_scope.select(:id).arel))
+      .else(arel_table[:client_id].in(client_scope.select(:id).arel))
+
+    viewable_scope = Hmis::File
+      .left_outer_joins(:client)
+      .left_outer_joins(:enrollment)
+      .where(case_statement)
+
+    viewable_scope = viewable_scope.or(Hmis::File.where(user_id: user.id)) if user.can_manage_own_client_files?
```

Comment:
Is can_manage_own_files supposed to be an extension of the files you can see, or is that "I can _only_ see my files."  In the warehouse, we tend to use that either as a limit, or to control which of the files you can see, you would be able to edit.

Reply:
The description is 'Grants the ability to manage user's own client files.' I just adapted the previous logic so I'm not sure but it looks additive, not limiting. @gigxz can you confirm?

Reply:
I think it should be additive. If you have "can manage own files" and "can view all non confidential files" then you should be able to see non confidential files including your own, but only be able to edit your own

Reply:
I think that's what this is doing then, right @ttoomey ?

Reply:
Yes, the condition above is additive. Essentially:
(user_is_owner) || ( enrollment ? (perm via enrollment) : (perm via client) )

There are file visibility tests here:
drivers/hmis/spec/requests/hmis/lookup_client_spec.rb

I can improve those tests to give more confidence in this implementation, sound good?

Reply:
I'm good with this.  Thanks!